### PR TITLE
add codeql-analysis workflow

### PR DESCRIPTION
**Pull request by chicco785**
## Proposed changes

Add codeql-analysis workflow

## Types of changes

What types of changes does your code introduce to the project: _Put
an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing
      functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after
creating the PR. If you're unsure about any of them, don't hesitate to
ask. We're here to help! This is simply a reminder of what we are going
to look for before merging your code._

- [x] I have read the [CONTRIBUTING][contrib] doc
- [x] I have signed the [CLA][cla]
- [x] I have added tests that prove my fix is effective or that my
      feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in
      downstream modules

## Further comments

N/A

[cla]: https://raw.githubusercontent.com/orchestracities/ngsi-timeseries-api/master/individual_cla.pdf
    "Martel Open Source Software Individual Contributor License Agreement"
[contrib]: https://github.com/smartsdk/ngsi-timeseries-api/blob/master/CONTRIBUTING.md
    "Contributing to QuantumLeap"

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
```
Comment (Cerfoglg) [inline .github/workflows/codeql-analysis.yml:7]: Should this comment remain?

File: RELEASE_NOTES.md [hunk 2]
```diff
@@ -14,6 +14,7 @@
 - Added a warning to use 'id' and 'type' from version 0.9 in all
    query responses (#584)
 - Added instanceId for each entry received (#565)
+- add codeql-analysis workflow
```
Comment (Cerfoglg) [inline RELEASE_NOTES.md:17]: *Added

**Review by Cerfoglg — APPROVED** [review]
Minor comments aside, looks good

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (chicco785) [reply .github/workflows/codeql-analysis.yml:7]: probably not :)

**Pull request merged**